openpype/hosts/maya/plugins/create/create_animation_pointcache.py

```diff
@@ -0,0 +1,317 @@
+                " the starting frame for time dependent translations and can"
+                " be used to evaluate run-up that isn't actually translated."
+            ),
+            disabled=True
+            if "preRollStartFrame" not in abc_args_overrides
+            else False,
+        )
+    )
+
+    return abc_defs
+
+
+def _ensure_defaults(cls, instance_data):
+    """Ensure we get default values when an attribute is not overrideable.
+
+    In instances where an attribute used to be modifiable, and then was locked
+    again, we want to make sure that we pass the default (what's on the
+    settings) instead of any value that might have been stored in the scene
+    when the attribute was modifiable.
+    """
+    abc_args_overrides = getattr(cls, "abc_args_overrides", [])
+    creator_attr = instance_data["creator_attributes"]
+    attr_default = getattr(cls, "attr", "")
+
+    if "attr" not in abc_args_overrides:
+        creator_attr["attr"] = attr_default
+
+    if "attrPrefix" not in abc_args_overrides:
+        creator_attr["attrPrefix"] = getattr(cls, "attrPrefix", "")
+
+    if "dataFormat" not in abc_args_overrides:
+        creator_attr["dataFormat"] = getattr(cls, "dataFormat", "")
+
+    if "preRollStartFrame" not in abc_args_overrides:
+        creator_attr["preRollStartFrame"] = getattr(
+            cls, "preRollStartFrame", ""
+        )
+
+    (
+        abc_boolean_defaults,
+        abc_boolean_overrides,
+    ) = _get_abc_export_flags(cls)
+
+    creator_attr["abcExportFlags"] = list(abc_boolean_defaults)
+
+    if creator_attr.get("abcExportTogglableFlags", []):
+        abc_boolean_args = creator_attr["abcExportTogglableFlags"].copy()
+
+        creator_attr["abcExportTogglableFlags"] = [
+            arg for arg in abc_boolean_args if arg not in abc_boolean_overrides
+        ]
+
+    return instance_data
+
+
+class CreateAnimation(plugin.MayaHiddenCreator):
+    """Animation output for character rigs
+
+    We hide the animation creator from the UI since the creation of it is
+    automated upon loading a rig. There's an inventory action to recreate it
+    for loaded rigs if by chance someone deleted the animation instance.
+    """
+
+    identifier = "io.openpype.creators.maya.animation"
+    name = "animationDefault"
+    label = "Animation"
+    family = "animation"
+    icon = "male"
+
+    write_color_sets = False
+    write_face_sets = False
+    include_parent_hierarchy = False
+    include_user_defined_attributes = False
+
+    def collect_instances(self):
+        try:
+            cached_subsets = self.collection_shared_data["maya_cached_subsets"]
+        except KeyError:
+            self.cache_subsets(self.collection_shared_data)
+            cached_subsets = self.collection_shared_data["maya_cached_subsets"]
+
+        for node in cached_subsets.get(self.identifier, []):
+            node_data = self.read_instance_node(node)
+            _ensure_defaults(self, node_data)
```

Comment:
`_ensure_defaults` takes `cls` as argument but here we're passing `self` which is the instance, not the class. This _may_ be confusing? Should the argument not be `cls`?